### PR TITLE
reorganised sub_data_update_tracerrelationships code ... 

### PR DESCRIPTION
**Pull request by derpycode**

### CODE REVIEW

**Comment by derpycode** [comment]
This needs careful checking!
It is simpler ... even if it does not look so, and more settings are more explicit (even if duplicating defaults).
The main real change was setting PON remin to NH4 under oxic conditions rather than NO3.
Everything else (in oxic remin) seemed is a confusing order/mess and I tried to address that.
In other redox sections, I only added an additional default behaviour / code comment.

File: genie-biogem/src/fortran/biogem_data.f90 [hunk 1]
```diff
@@ -971,54 +971,93 @@ SUBROUTINE sub_data_update_tracerrelationships()
     ! -------------------------------------------------------- !
     ! UPDATE REDFIELD RELATIONSHIPS
```
Comment (FannyMonteiro) [inline genie-biogem/src/fortran/biogem_data.f90]: Would say
UPDATE REDFIELD RELATIONSHIPS DURING REMINERALISATION

File: genie-biogem/src/fortran/biogem_data.f90 [hunk 2]
```diff
@@ -971,54 +971,93 @@ SUBROUTINE sub_data_update_tracerrelationships()
...
     ! -------------------------------------------------------- !
-    ! if NO3 is employed;
-    ! calculate alkalnity corrections associated with the formation and destruction of organic matter from NO3
-    ! otherwise, convert PO4 units to NO3 via the P:N Redfield ratio and then calculate the ALK correction from NO3
+    ! N
```
Comment (FannyMonteiro) [inline genie-biogem/src/fortran/biogem_data.f90]: Would say here:
Code update for PON remineralisation

File: genie-biogem/src/fortran/biogem_data.f90 [hunk 3]
```diff
@@ -971,54 +971,93 @@ SUBROUTINE sub_data_update_tracerrelationships()
...
+    ! NOTE: some of the zero values here are already the (hard-coded) defaults ... they are repeated here to be completely explicit
+    if (ocn_select(io_NH4)) then
+       conv_sed_ocn(io_ALK,is_PON) = conv_sed_ocn(io_NH4,is_PON)
+       conv_ocn_sed(is_PON,io_ALK) = 1.0/conv_sed_ocn(io_ALK,is_PON)
+       conv_sed_ocn(io_ALK,is_POP) = 0.0
+       conv_ocn_sed(is_POP,io_ALK) = 0.0
+       conv_sed_ocn(io_ALK,is_POC) = 0.0
+       conv_ocn_sed(is_POC,io_ALK) = 0.0
+    elseif (ocn_select(io_NO3)) then
+       conv_sed_ocn(io_ALK,is_PON) = -conv_sed_ocn(io_NO3,is_PON)
```
Comment (FannyMonteiro) [inline genie-biogem/src/fortran/biogem_data.f90]: The correction for Alk with NH4 looks good to me. I'm not sure about the correction when only NO3 is selected. My understanding is that Alk = HCO3- + 2 CO32- + B(OH)4- + OH- - H+ + HPO42- + 2PO43- + H3SiO4- + NH3 +HS- ...

When we remineralise PON into NO3, we assume that nitrification is instantaneous: PON->NH3->NO3-. Looking at the full equation for nitrification we get:
NH3 + O2 + H2O -> NO3- + 5H+ + 4e-

So do we not need to remove some Alk due to 5H+ production?

File: genie-biogem/src/fortran/biogem_data.f90 [hunk 4]
```diff
@@ -971,54 +971,93 @@ SUBROUTINE sub_data_update_tracerrelationships()
...
+    if (ctrl_bio_red_O2withPOC) then
+       conv_sed_ocn(io_O2,is_POP) = 0.0
+       conv_ocn_sed(is_POP,io_O2) = 0.0
+    else
+       conv_sed_ocn(io_O2,is_POP) = -4.0/2.0
+       conv_ocn_sed(is_POP,io_O2) = 1.0/conv_sed_ocn(io_O2,is_POP)
+    endif
+    if (ocn_select(io_NH4)) then
+       conv_sed_ocn(io_O2,is_PON) = (3.0/4.0)
+       conv_ocn_sed(is_PON,io_O2) = -(4.0/5.0)
```
Comment (FannyMonteiro) [inline genie-biogem/src/fortran/biogem_data.f90]: What to do about phytoplankton uptake of NH4? conv_ocn_sed(is_PON,io_O2) should be zero as NH4+ -> PON + 4H+

File: genie-biogem/src/fortran/biogem_data.f90 (hunk 1 above)
Comment (derpycode) [reply genie-biogem/src/fortran/biogem_data.f90]: [will improve commenting in a future merge]

File: genie-biogem/src/fortran/biogem_data.f90 (hunk 2 above)
Comment (derpycode) [reply genie-biogem/src/fortran/biogem_data.f90]: [will improve commenting in a future merge]

**Pull request merged**